src/lib/util/Config.ts

```diff
@@ -193,10 +209,18 @@ export class Config {
 		return this.getStringArrayConfigValue('targetPatterns', engine);
 	}
 
+	public getMinimumTokens(engine: ENGINE): Promise<number> {
```

Comment:
It seems like this class should contain generic methods that aren't specific to a single engine. The code is somewhat inconsistent in this regard with `getSupportedLanguages`, but it's possible that the `supportedLanguages` attribute could be added to other engines later.

Could this method be moved into the CPD engine and the `getNumberConfigValue` value be made public? @rmohan20 interested on your thoughts as well.

Reply:
@jbartolotta-sfdc Sorry, I missed this earlier.
Having `getMinimumTokens()` on config does feel weird in terms of other engines that don't have any context about this value. But do we want CpdEngine to know config property names? Wouldn't that be config format leaking into the engine class?
Slightly related general problem is that we force all the engines to have the same format of fields in config.

Reply:
I think it can stay where it is.

The Config class currently has a lot of knowledge about engine internals. Until we make this class more modular, it makes sense to leave the method here. Eventually it would be good to get to a place where the Config class is more modular and the keys are only known by the caller interested in the value.

I think Config's responsibilities should be loading, saving and enforcing type safety but not necessarily have engine specific knowledge. This is the architecture of the sfdx config plugin, the config plugin is responsible for general use storage of config but doesn't have specific knowledge of what is stored.